amazon_kclpy/kcl.py

```diff
@@ -220,6 +220,17 @@ def shutdown(self, checkpointer, reason):
         '''
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def shutdownRequested(self, checkpointer):
```

Comment:
This should follow [Python PEP8](https://www.python.org/dev/peps/pep-0008/#function-names) style e.g. shutdown_requested.

For the v2 interface this should also use intermediate type, even though I did that wrong with the interface in the Java KCL.  See [v2/processor.py#L51](https://github.com/awslabs/amazon-kinesis-client-python/blob/master/amazon_kclpy/v2/processor.py#L51) for an example